amlip_py/amlip_py/types/StatusDataType.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""AML-IP Status data type API specification."""
+
+from enum import Enum
+
+from amlip_py.types.AmlipIdDataType import AmlipIdDataType
+
+from amlip_swig import NodeKind_agent, NodeKind_computing, NodeKind_discovery, NodeKind_main
+from amlip_swig import NodeKind_meta, NodeKind_status, NodeKind_undetermined
+from amlip_swig import StateKind_dropped, StateKind_running, StateKind_stopped, StateKind_unknown
+from amlip_swig import StatusDataType as cpp_StatusDataType
+
+
+class NodeKind(Enum):
+    """Enumeration to represent the different Node Kinds of each Node."""
+
+    UNDETERMINED = NodeKind_undetermined
+    DISCOVERY = NodeKind_discovery
+    AGENT = NodeKind_agent
+    MAIN = NodeKind_main
+    COMPUTATIONAL = NodeKind_computing
```

Comment:
We should use the same name everywhere I think.

Reply:
Probably. But C++ and python conventions differ. And C++ lowercase enumerations make sense with enum class because it is forced to use the namespace, but in Python it does not exist.